### DISCUSSION TITLE
Importing tailwind config file in a Create-React-App breaks while attempting a production build

### DISCUSSION THREAD

**Discussion by rinajhinjhina** [tailwindlabs/tailwindcss#1853] [Help]
Hi, first of all thank you for all your work on this wonderful project.

I faced this particular edge case while importing the tailwind config so I could extract breakpoint variables using `resolveConfig` according the [referencing config in Javascript guide](https://tailwindcss.com/docs/configuration/#referencing-in-javascript).

```js
import tailwindConfig from "../../../tailwind.config";
import resolveConfig from "tailwindcss/resolveConfig";

const config = resolveConfig(tailwindConfig);

const mdBreakpoint = config.theme.screens.md
```

While this works just fine in development, it breaks when I run `yarn build` with this error:

```
Failed to compile.

./src/core/utils/tailwindUtils/tailwindTheme.ts
Attempted import error: '../../../tailwind.config' does not contain a default export (imported as 'tailwindConfig').
```

I apologize if the problem is with CRA and not tailwind but I definitely would like to get to the bottom of this.

**Comment by estevanmaito** [comment]
I think it has something to do with Webpack seeing mixed `import` and `module.exports`

What happens if you change your imports to `require`s?

```js
const tailwindConfig = require("../../../tailwind.config");
const resolveConfig = require("tailwindcss/resolveConfig");
```

*considering that both files export using `module.exports`

**Comment by rinajhinjhina** [comment]
It then compiles fine but throws a runtime error.

```
TypeError: "exports" is read-only
```

**Comment by amille14** [comment]
I'm having this same issue. Did you ever find a solution? Currently getting `Uncaught TypeError: Cannot assign to read only property 'exports' of object '#<Object>'` in my `tailwind.config.js`

**Reply by kevinbischof** [reply]
Actually I'm facing the same issue. No help for this problem?

**Comment by sara-jegorova** [comment]
I fixed this issue by adding babel-plugin-preval plugin (using craco to avoid ejecting CRA) and then did:

```
const fullConfig = preval`
  const resolveConfig = require('tailwindcss/resolveConfig');
  const tailwindConfig = require('../../tailwind.config');
  module.exports = resolveConfig(tailwindConfig);
`;
```

This keeps the bundle size smaller, and removes the problem of accessing the config outside of `src` folder. Hope this helps!

**Reply by marabesi** [reply]
can you provide a full example? Where did you place this snippet?

**Reply by marabesi** [reply]
I tried to share what I did here https://marabesi.com/web/2021/10/31/sharing-tailwind-config-elsewhere-for-variable-access.html

btw I didn't like the solution, but at least I could unblock myself.

**Reply by doubleedev** [reply]
Is the solution complete? It doesn't say what to do after creating the tailwindWrapper?

**Reply by marabesi** [reply]
you are correct, just updated the solution to be more explicit on what I did and how to use that (there is a jsx component that I use the solution proposed)
